xarray/core/extension_array.py

```diff
@@ -102,7 +102,7 @@ def replace_duck_with_extension_array(args) -> list:
             return type(self)[type(res)](res)
         return res
 
-    def __array_ufunc__(ufunc, method, *inputs, **kwargs):
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
```

Comment:
the test failure in `test_units` has to do with this implementation for the equals ufunc.

@ilan-gold can you take a look please?

Reply:
The signature is correct https://numpy.org/devdocs/user/basics.subclassing.html

I have seen that error before but I can't remember what it means now, maybe a segfault...

Reply:
I saw that, but we have `self` included for `__array_ufunc__` inside Xarray 🤷🏾‍♂️

Reply:
ok! hopefully my fix helps, but it was hard to make heads or tails of what changed because the diff was super dirty (there was a merge from `main` or two).  please open a PR into mine in the future, would make things easier :)